client/query_impl.go

```diff
@@ -0,0 +1,127 @@
+package client
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/iov-one/tendermint/rpc/client"
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/app"
+	"github.com/iov-one/weave/errors"
+	abci "github.com/tendermint/tendermint/abci/types"
+)
+
+func NewRPCABCIQuery(c client.Client) Query {
+	return &RPCABCIQuery{c}
+}
+
+// RPCABCIQuery exposes query interface via tendermint rpc
+type RPCABCIQuery struct {
+	conn client.Client
+}
+
+func (q *RPCABCIQuery) Get(key []byte) []byte {
+	query, err := q.conn.ABCIQuery("/", key)
+	if err != nil {
+		panic(err)
```

Comment:
req: we should not make assumptions how this client package is used. Returning the error is the best practice in Go unless you can provide it some other way.
Please revisit all panics!